lib/openactive/rpde/rpde_body.rb

```diff
@@ -0,0 +1,143 @@
+      # @option opts [String]  feed_base_url string
+      # @option opts [Integer]  modified int
+      # @option opts [String,Integer]  id string|int
+      # @option opts [Array<::OpenActive::Rpde::RpdeItem>] items
+      # @return self
+      # @throws [::OpenActive::Rpde::Exceptions::FirstTimeAfterTimestampAndAfterIdException] If the afterId and afterTimestamp provided are of the first item in the feed
+      # @throws [::OpenActive::Rpde::Exceptions::DeletedItemsDataException] If any deleted items provided (if any) contain data
+      # @throws [::OpenActive::Rpde::Exceptions::IncompleteItemsDataException] If any RPDE feed item does not include id, modified, state, or kind.
+      # @throws [::OpenActive::Rpde::Exceptions::ModifiedIdItemsOrderException] If the feed items are not in "modified", then "id", order.
+      # @throws [::OpenActive::Exceptions::InvalidArgumentException] If the provided argument is not of a supported type.
+      def self.create(feed_base_url:, modified: nil, id: nil, change_number: nil, items:)
+        if !modified.nil? && !id.nil? && change_number.nil?
+          self.create_from_modified_id(feed_base_url, modified, id, items)
+        elsif modified.nil? && id.nil? && !change_number.nil?
+          self.create_from_next_change_number(feed_base_url, change_number, items)
+        else
+          raise ArgumentError
+        end
+      end
+
+      # Create an RpdeBody object from a given feed base URL, a modified int,
+      # an id, and an array of RpdeItem.
+      # Checks are performed to see if the provided "modified" and "id" attributes
+      # are valid.
+      #
+      # @param [String] feed_base_url
+      # @param [Integer] modified int
+      # @param [String,Integer] id
+      # @param [Array<::OpenActive::Rpde::RpdeItem>] items
+      # @return self
+      # @throws [::OpenActive::Rpde::Exceptions::FirstTimeAfterTimestampAndAfterIdException] If the afterId and afterTimestamp provided are of the first item in the feed
+      # @throws [::OpenActive::Rpde::Exceptions::DeletedItemsDataException] If any deleted items provided (if any) contain data
+      # @throws [::OpenActive::Rpde::Exceptions::IncompleteItemsDataException] If any RPDE feed item does not include id, modified, state, or kind.
+      # @throws [::OpenActive::Rpde::Exceptions::ModifiedIdItemsOrderException] If the feed items are not in "modified", then "id", order.
+      # @throws [::OpenActive::Exceptions::InvalidArgumentException] If the provided argument is not of a supported type.
+      def self.create_from_modified_id(feed_base_url, modified, id, items)
+        rpde_body = self.new(
+          items: items
+        )
+        if items.length > 0
+          first_item = items.first
+
+          # Checks that the afterId and afterTimestamp provided are not the
+          # first item in the feed (helps detect whether query is correct)
+          if first_item.modified.to_s == modified.to_s && first_item.id.to_s == id.to_s
```

Comment:
`(first_item.modified.to_s < modified.to_s) || (first_item.modified.to_s == modified.to_s && first_item.id.to_s <= id.to_s)` maybe?

Reply:
Wouldn't work due to being strings now, would need to attempt to parse them.

Also, in it's current state it'sa identical to the PHP and .NET implementations:
- https://github.com/openactive/models-php/blob/master/src/Rpde/RpdeBody.php#L99-L104
 - https://github.com/openactive/OpenActive.NET/blob/master/OpenActive.NET/rpde/RDPEModels.cs#L148-L150